internal/jobobject/jobobject.go

```diff
@@ -503,6 +507,27 @@ func (job *JobObject) ApplyFileBinding(root, target string, merged bool) error {
 	return nil
 }
 
+// isOpenedJobSilo is a helper to determine if a job object that was opened is a silo. This should ONLY be called
+// from `Open` and any callers in this package afterwards should use `job.isSilo()``
+func (job *JobObject) isOpenedJobSilo() bool {
+	// None of the information from the structure that this info class expects will be used, this is just used as
+	// the call will fail if the job hasn't been upgraded to a silo so we can use this to tell when we open a job
+	// if it's a silo or not. Because none of the info matters simply define a dummy struct with the size that the call
+	// expects which is 16 bytes.
+	type isSiloObj struct {
+		_ [16]byte
+	}
+	var siloInfo isSiloObj
+	err := winapi.QueryInformationJobObject(
+		job.handle,
+		winapi.JobObjectSiloBasicInformation,
+		uintptr(unsafe.Pointer(&siloInfo)),
+		uint32(unsafe.Sizeof(siloInfo)),
+		nil,
+	)
+	return err == nil
```

Comment:
Also, this may error is the handle is invalid or we dont have access to it, right?
If so, can we change the return to `bool, error`, and return the error in non-silo cases?

Reply:
We'd have to have access if Open worked which is the only place this is called. This is kind of relying a bunch on being only called after opening a job..